### PR TITLE
[JENKINS-73951][JENKINS-73952] Improve CSP compatibility

### PR DESCRIPTION
**Pull request by yaroslavafenkin**
https://issues.jenkins.io/browse/JENKINS-73951
https://issues.jenkins.io/browse/JENKINS-73952

### Testing done
For JENKINS-73951 I needed to go to the `enterCredential` page that's referenced from `url` field help file. And then to get to the `credentialOK` I commented out these lines locally: https://github.com/jenkinsci/subversion-plugin/blob/c73117ed289a1631041aae8b7246cffe8ac30275/src/main/java/hudson/scm/SubversionSCM.java#L2347-L2348

https://www.loom.com/share/4b2e8bfe5ab147e8b05584dded1343f1

https://www.loom.com/share/94b12cc934c8439f879bf5c8cc6709eb

CSP violations that appear in the list in the end are addressed by https://github.com/jenkinsci/subversion-plugin/pull/319.

---

For JENKINS-73952 had to setup a project with 2 module locations, so that at least 2 entries appear on the `tagBuild` page which allows affected elements to be rendered. The functionality in the related script was broken due to jQuery not being available, so I've fixed that along the way.

https://www.loom.com/share/2b96f17fe29340ec8d536ebedbd9517e

https://www.loom.com/share/58b6fdaae7384c86bf9dd64b0ebe93f0

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/hudson/scm/SubversionSCM/DescriptorImpl/credential-ok.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+window.addEventListener("DOMContentLoaded", () => {
+    document.querySelector(".svn-credential-ok-close").addEventListener("click", () => window.close());
```
Comment (basil) [inline src/main/resources/hudson/scm/SubversionSCM/DescriptorImpl/credential-ok.js:2]: Elsewhere in your changes, you've been putting one-line lambdas into their own block. I don't care either way, as long as we're consistent within the same file.

```suggestion
    document.querySelector(".svn-credential-ok-close").addEventListener("click", () => {
        window.close();
   });
```

File: src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+function updateRow(changeEvent) {
+    const checkbox = changeEvent.target;
+    checkbox.parentNode.parentNode.style.color = checkbox.checked ? "inherit" : "grey";
+
+    const index = checkbox.getAttribute("data-index");
+    document.querySelector(`input[name="name${index}"]`).disabled = !checkbox.checked;
+}
+
+window.addEventListener("DOMContentLoaded", () => {
+    document.querySelectorAll(".svn-tagform-tag-checkbox").forEach(checkbox => {
```
Comment (basil) [inline src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js:11]: Elsewhere in Jenkins core and in other CSP PRs we're putting parameters in parentheses consistently, so please do so here as well:

```suggestion
    document.querySelectorAll(".svn-tagform-tag-checkbox").forEach((checkbox) => {
```

File: src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        document.addEventListener("change", updateRow);
```
Comment (basil) [inline src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js:12]: It seems odd to me that we aren't using `checkbox`. If it's truly unused, maybe we should use `()` instead on the previous line? Or if we should use it, should this line be:

```suggestion
        checkbox.addEventListener("change", updateRow);
```

**Review by basil — APPROVED** [review]
Testing looks good. Left some optional feedback for your consideration. Feel free to disregard if these suggestions aren't helping.

File: src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js (hunk 3 above)
Comment (yaroslavafenkin) [reply src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js:12]: Oh, it's an oversight from my side. I'm unsure how this even worked for the sake of the video :/

File: src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js (hunk 2 above)
Comment (yaroslavafenkin) [reply src/main/resources/hudson/scm/SubversionTagAction/svn-tagform-update-row.js:11]: Thanks, I'll try to be more consistent with it.

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Pull request merged**